### ISSUE TITLE
Figure out how test toolsheds work

### ISSUE THREAD

**Issue by ebolyen** [qiime2/galaxy-tools#10]
We have a plugin called [`q2-mystery-stew`](https://github.com/qiime2/q2-mystery-stew) which produces the tests used by q2galaxy.

Ideally we could render this plugin and start testing toolshed server to install this suite and run the tests. This would also help prove that our `.shed.yml` files are meaningful.

There's probably a lot of overlap here with #7

**Comment by ebolyen** [comment]
A question, is there any conventions for a tools directory which is either only for testing, or is somehow excluded from uploading?

I would prefer Galaxy users not interact with q2-mystery-stew directly on the toolshed, as it's literally a combinatoric soup of nonsense.

**Comment by bernt-matthias** [comment]
The CI implemented in https://github.com/galaxyproject/galaxy-tool-repository-template (which is used at IUC and essentially every other Galaxy tool repository) currently does not allow to skip deployment of certain tools. But this should be easy to implement. 

Ping me if you think this is needed. 

> start testing toolshed server to install this suite and run the tests

Note that the toolshed is not running any tool tests. Tool tests of all repos run via github actions:

-  for PRs https://github.com/galaxyproject/galaxy-tool-repository-template/blob/main/.github/workflows/pr.yaml
- and weekly for all tools https://github.com/galaxyproject/galaxy-tool-repository-template/blob/main/.github/workflows/ci.yaml

Furthermore there is the possibility to run slash commands in issues to test against certain Galaxy branches (which might be nice to check tools against a Galaxy PR, e.g. an upcoming datatype addition PR)

**Comment by bernt-matthias** [comment]
> Another option might be to have a parallel test branch with an alternate CI configuration, so that upload is avoided, but tests still run.

We already have a files `.tt_skip` and `.tt_biocontainer_skip` to control tools / directories that should not be subject to CI. So it should be really easy to setup another one to forbid deployment. But another branch also sounds useful .. up to you :)

> These github actions are really detailed! It does look like it uses planemo test under the hood. 

Indeed. Everything works with this action https://github.com/galaxyproject/planemo-ci-action which uses `planemo`.

> This means there isn't some mock toolshed which a galaxy instance is fetching from and then testing? 

Exactly.

> If that's the case, then testing with mystery-stew can probably just continue to live in q2galaxy and we don't need to do anything special in this repo. [This is what we do at the moment](https://github.com/qiime2/q2galaxy/blob/master/Makefile#L21-L29).
> Then this particular repo can look like any of the other ones.

Sound reasonable. Guess the tool tests derived from [Usage Examples](https://dev.qiime2.org/latest/api-reference/usage/) will end up here, or?

**Comment by ebolyen** [comment]
> Sound reasonable. Guess the tool tests derived from [Usage Examples](https://dev.qiime2.org/latest/api-reference/usage/) will end up here, or?

Yep, any examples that do exist (with ideally more in the future) will become test sections and should be executed here.
